portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -34,18 +34,52 @@
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+
 import com.google.sps.data.Comment;
+import com.google.sps.data.UserInfo;
 
 /** Servlet responsible for deleting comments. */
 @WebServlet("/delete-comment")
 public class DeleteCommentServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    response.setContentType("application/json");
 
-    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+    long id = Long.parseLong(request.getParameter("id"));
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(taskEntityKey);
+    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+
+    //Only delete a comment if the author is trying to delete it. 
+    try {
+      String commentUserId = (String) datastore.get(taskEntityKey).getProperty("userId");
+
+      UserService userService = UserServiceFactory.getUserService();
+      UserInfo userInfo = new UserInfo();
+
+      String currentUserId = userService.getCurrentUser().getUserId();
+      userInfo.userId = currentUserId;
+
+      if (userService.isUserLoggedIn()) {
+        if(currentUserId.equals(commentUserId)) {
+          datastore.delete(taskEntityKey);
```

Comment:
What happens when the comment doesn't exist?

Reply:
I believe this will be handled by the code '''    } catch (com.google.appengine.api.datastore.EntityNotFoundException enfe) {'''

Reply:
Specifically it won't be able to create/find a corresponding taskEntityKey and will break.